libs/blocks/youtube/youtube.js

```diff
@@ -14,12 +14,23 @@ export default function init(a) {
         webkitallowfullscreen mozallowfullscreen allowfullscreen
         allow="encrypted-media; accelerometer; gyroscope; picture-in-picture"
         scrolling="no"
+        id="player-${id}"
         title="${title}">
       </iframe>
     </div>`;
     a.insertAdjacentHTML('afterend', embedHTML);
     a.remove();
+    if (document.readyState === 'complete') {
+      /* eslint-disable-next-line no-underscore-dangle */
+      window._satellite?.track('trackYoutube');
+    } else {
+      document.addEventListener('readystatechange', () => {
+        if (document.readyState === 'complete') {
+          /* eslint-disable-next-line no-underscore-dangle */
+          window._satellite?.track('trackYoutube');
```

Comment:
Should a named callback be used so the event listener can be removed?

Reply:
I was going to suggest the `once` approach, but now I have another question - can we have multiple YouTube videos on a page? If so, would all videos be considered by this satellite call?

Reply:
Pasting Atanu's comment who requested this functionality:
<img width="759" alt="image" src="https://github.com/adobecom/milo/assets/110388864/60f0aa51-2b98-4baf-80b0-5f7c43784c65">

Reply:
@overmyheadandbody : Does this answer your request? Is there anything else you need from my side?

Reply:
Hey @suhjainadobe! PR was approved from my side; this comment shows that follow-ups may be needed in the future since this seems to be a POC

Reply:
Yes, we may need to refer it in future. Merging the PR as it is needed by the analytics team.